src/MoBi.Engine/Sbml/UnitDefinitionImporter.cs

```diff
@@ -146,45 +155,48 @@ public IDimension ConvertUnit(UnitDefinition unitDefinition)
                sbmlUnitDefinition.getScale(),
                _baseUnitsDictionary);
          }
+
          var unitName = combinedUnit.Name;
-         var dimensionAndUnit = _dimensionFactory.FindUnit(unitName);
+         var dimensionAndUnit = _moBiDimensionFactory.FindUnit(unitName);
          if (dimensionAndUnit.dimension != Constants.Dimension.NO_DIMENSION)
          {
             _sbmlInformation.MobiDimension[sbmlUnit] = dimensionAndUnit.dimension;
-            _unitConvertionDictionary.Add(sbmlUnit, new UnitConvertionInfo() { Dimension = dimensionAndUnit.dimension, Unit = dimensionAndUnit.unit, Rate = combinedUnit.Rate });
+            _unitConversionDictionary.Add(sbmlUnit, new UnitConversionInfo {Dimension = dimensionAndUnit.dimension, Unit = dimensionAndUnit.unit, Rate = combinedUnit.Rate});
             return dimensionAndUnit.dimension;
          }
 
-         _unitConvertionDictionary.Add(sbmlUnit, new UnitConvertionInfo() { Dimension = dimension, Unit = dimension.DefaultUnit, Rate = 1 });
+         _unitConversionDictionary.Add(sbmlUnit, new UnitConversionInfo {Dimension = dimension, Unit = dimension.BaseUnit, Rate = 1});
          return dimension;
       }
 
-      public override void AddToProject() { }
+      public override void AddToProject()
+      {
+      }
 
       public IDimension DimensionFor(string sbmlUnit)
       {
-         if (_unitConvertionDictionary.ContainsKey(sbmlUnit))
-            return _unitConvertionDictionary[sbmlUnit].Dimension;
+         if (_unitConversionDictionary.ContainsKey(sbmlUnit))
+            return _unitConversionDictionary[sbmlUnit].Dimension;
 
          return _moBiDimensionFactory.DimensionForUnit(TranslateUnit(sbmlUnit)) ?? _moBiDimensionFactory.NoDimension;
       }
 
       public (double value, IDimension dimension) ToMobiBaseUnit(string unit, double value)
       {
-         if (_unitConvertionDictionary.ContainsKey(unit))
+         if (_unitConversionDictionary.ContainsKey(unit))
          {
-            var convertionData = _unitConvertionDictionary[unit];
-            return (convertionData.Dimension.UnitValueToBaseUnitValue(convertionData.Unit, value * convertionData.Rate), convertionData.Dimension);
+            var conversionData = _unitConversionDictionary[unit];
+            return (conversionData.Dimension.UnitValueToBaseUnitValue(conversionData.Unit, value * conversionData.Rate), conversionData.Dimension);
```

Comment:
Why do we need this rate? It feels that yo are converting twice since we are also converting using the value

Reply:
@msevestre There are indeed two conversions. The first conversion is based on the description on the sbml file. The second is to the mobi base unit. Suppose there is a description from unit A to a (sbml) base unit B described in the file. This will be captured by the rate. Unit A does not exist on mobi but unit B does. Additionally, unit B needs to be translated to mobi base unit C. More specifically, we have an entry on the sbml file that specifies that "unit A" is going to be used and it has a given rule to be translated to "mole" (which is a conventional unit known in mobi too). This translation is captured in rate. But when we translate in mobi, we actually need to translate from "unit A" to "micro mole" which is the base unit, so we need to translate from "unit A" to "mole" and then to "micro mole". Does it make sense? It is not a clean explanation but we could talk about this tomorrow if you like. I will take a look on the changes though and also on the failing test.